include/ofi.h

```diff
@@ -158,17 +158,14 @@ static inline int ofi_val32_ge(uint32_t x, uint32_t y) {
 #define TAB "    "
 
 #define CASEENUMSTR(SYM) \
-	case SYM: { ofi_strcatf(buf, #SYM); break; }
+	case SYM: { ofi_strncatf(buf, buflen, #SYM); break; }
 #define IFFLAGSTR(flags, SYM) \
-	do { if (flags & SYM) ofi_strcatf(buf, #SYM ", "); } while(0)
+	do { if (flags & SYM) ofi_strncatf(buf, buflen, #SYM ", "); } while(0)
 #define CASEENUMSTRN(SYM, N) \
 	case SYM: { ofi_strncatf(buf, N, #SYM); break; }
 #define IFFLAGSTRN(flags, SYM, N) \
 	do { if (flags & SYM) ofi_strncatf(buf, N, #SYM ", "); } while(0)
```

Comment:
I completely missed last review that we already have the macros that use ofi_strcatf.  We should probably have a follow on patch to remove the 'STR' macros and replace them with the 'STRN' versions that already exist.  Plus I don't really like the way the macros are defined, since they assumed local variables exist with a specific name.